UI for onboarding flow

Elanhant opened the pull request:
This PR adds pages and components for most of the onboarding flow. No API calls or anything, with the exception of calling `useSites` to display a list of detected sites.

To get to the onboarding flow, you can use a temporary "Onboarding" button on the homepage:
<img width="879" alt="Screen Shot 2020-08-27 at 12 48 09 PM" src="https://user-images.githubusercontent.com/4366711/91488077-9bb5a300-e863-11ea-8727-00e11ff391ce.png">

Elanhant committed:
UI for onboarding flow

Elanhant committed:
Fix editor identifiers

Elanhant committed:
Polish site card UI

Elanhant committed:
Wire up 'Add a site' to onboarding

ascorbic reviewed (commented):
![image](https://user-images.githubusercontent.com/213306/91726676-8c529480-eb98-11ea-8faf-d9b737d2fcd4.png)
![image](https://user-images.githubusercontent.com/213306/91726706-94123900-eb98-11ea-994c-8f3b57c0f525.png)

I don't know if this is due to an existing issue, but the layout doesn't handle smaller window sizes very well. This is at the default window size.

Elanhant commented:
> I don't know if this is due to an existing issue, but the layout doesn't handle smaller window sizes very well. This is at the default window size.

Interesting, I think it was working fine for me, but I'll make sure to check again!

Elanhant committed:
Use EmptyState for no sites

Elanhant committed:
Fix responsiveness issues

Elanhant commented:
Onboarding flow should work better now on smaller screens (turns out having `position: absolute` on that SVG graphic was the culprit, I changed it to `position: fixed`).

ascorbic commented on src/components/folder-name.tsx:
```diff
@@ -24,7 +24,9 @@ export function FolderName({ sitePath }: IProps): JSX.Element {
     >
-      <MdFolderOpen size={12} /> <span sx={{ paddingLeft: 2 }}>{folder}</span>
+      <MdFolderOpen size={12} aria-label="Folder:" />
+      {` `}
```
I think this is a Prettier artefact

ascorbic commented on src/components/site-card/logs-viewer.tsx:
```diff
@@ -0,0 +1,42 @@
+      <StyledPanelHeader onCloseButtonClick={onDismiss}>Logs</StyledPanelHeader>
+      <RawLogs
+        // RawLogs expect "message" to be a string but in factthey also support elements
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```
So this is a workaround for bad typings in gatsby-interface? Can this be fixed there?

ascorbic commented:
This fixes the horizontal scroll, but there's still a vertical scroll at the default window size. I think this is because the padding causes the grid to switch to single column at the default window size. Could this be fixed?
<img width="925" alt="image" src="https://user-images.githubusercontent.com/213306/91826429-7941d200-ec35-11ea-859c-53ff68ddb5da.png">

ascorbic committed:
Open links from admin in browser

ascorbic commented on src/pages/onboarding/[step].tsx:
```diff
@@ -0,0 +1,60 @@
+          fontFamily: `sans`,
+        }}
+      >
+        {step === `editor` && (
```
I'm not convinced of the value of using unified routes in this situation. There are a fixed set of steps, each with its own component. I'd rather create a wrapper component to share, and a page template for each step

ascorbic committed:
Wire up preferred editor

ascorbic committed:
Persist telemetry preference

ascorbic committed:
WIre up choosing sites

ascorbic committed:
Merge branch 'feat/full-window' into Elanhant/onboarding-ui

ascorbic committed:
Merge branch 'Elanhant/onboarding-ui' into feat/wire-up-prefs

Elanhant committed:
Merge pull request #21 from gatsbyjs/feat/wire-up-prefs

Wire up onboarding settings

Elanhant committed:
Address updates in Figma

Elanhant committed:
Create onboarding step pages

Elanhant commented:
> This fixes the horizontal scroll, but there's still a vertical scroll at the default window size. I think this is because the padding causes the grid to switch to single column at the default window size. Could this be fixed?

We should probably consult with design on this. I also have a feeling that our current default size is too small, perhaps we can just increase it if possible?

ascorbic commented:
> We should probably consult with design on this. I also have a feeling that our current default size is too small, perhaps we can just increase it if possible?

I agree, but I do think the responsive behaviour could be improved, so that the top and left padding decreases. I'm happy to levae it for now, and will increase the default size so that it fits iniitally.

ascorbic committed:
Rearrange pages
